cluster-state-service/handler/store/datastore.go

```diff
@@ -110,13 +112,25 @@ func (datastore etcdDataStore) Get(key string) (map[string]string, error) {
 }
 
 // StreamWithPrefix starts a go routine that streams key-value pairs whose keys start with keyPrefix into the channel returned
-func (datastore etcdDataStore) StreamWithPrefix(ctx context.Context, keyPrefix string) (chan map[string]string, error) {
+func (datastore etcdDataStore) StreamWithPrefix(ctx context.Context, keyPrefix string, entityVersion string) (chan map[string]types.Entity, error) {
 	if len(keyPrefix) == 0 {
 		return nil, errors.New("Key prefix cannot be empty while streaming data from datastore by prefix")
 	}
 
-	kvChan := make(chan map[string]string) // go routine datastore.stream() handles closing of this channel
-	go datastore.stream(ctx, keyPrefix, kvChan)
+	// If entity version is specified, verify that it is not out of range.
+	if entityVersion != "" {
+		var revision int64
+		var err error
+		if revision, err = strconv.ParseInt(entityVersion, 10, 64); err != nil {
```

Comment:
The reason I can not use regex.IsEntityVersion() here is because I actually need the int returned by the ParseInt call, whereas regex.IsEntityVersion just returns a boolean.